### PR TITLE
Because I could

### PR DESCRIPTION
**Pull request by cclauss**

### CODE REVIEW

File: pythonista/shared.py [hunk 1]
```diff
@@ -39,4 +17,4 @@
 consoleVC = app.delegate().consoleViewController()
```
Comment (The-Penultimate-Defenestrator) [inline pythonista/shared.py]: I'm sorry, could you explain this? It's a little above my level…

**Comment by controversial** [comment]
Could you explain this a little bit more? I don't doubt its awesomeness, just wondering what it does & how it works.

**Comment by controversial** [comment]
If you could, it'd also be useful to me to have an explanation of dglsseus's code, as I didn't read it, and I'm just not quite sure what it does. I'd like, if possible, to stay on top of understanding all the code in this module.

**Comment by cclauss** [comment]
@dgelessus code is a super slick use of set theory.  Sets are very useful in logic, math and computer science for applying [DRY](https://en.m.wikipedia.org/wiki/Don%27t_repeat_yourself) to datasets.  Having [sets](https://docs.python.org/2/tutorial/datastructures.html#sets) built into Python is great but they are too often underutilized because folks struggle with the concepts of unions, intersections, two kinds of differences, two kinds of subsets, two kinds of supersets and strange operator precedence but these ideas are [not so difficult](https://github.com/cclauss/Ten-lines-or-less/blob/master/set_theory.py) and is surprisingly powerful.

So @dgelessus code is creating three sets and is taking the Set Difference between the second and third and then taking the Union of that result and the first.

``` python
print(sorted( {'a', 'b', 'c'} | {'c', 'd', 'e'}))                # --> ['a', 'b', 'c', 'd', 'e']
print(sorted( {'a', 'b', 'c'} | {'c', 'd', 'e'}  - {'b', 'd'}))  # --> ['a', 'b', 'c', 'e']
print(sorted(({'a', 'b', 'c'} | {'c', 'd', 'e'}) - {'b', 'd'}))  # --> ['a', 'c', 'e']  Note: `-` has higher operator precedence than `|`.
```

My code just uses the parlor trick of combining multiline strings with str.split() to squeeze @dgelessus code into just four lines with a somewhat startling loss of clarity which is why I called this pull request "Because I can" and I will not be at all surprised or offended if you reject it ;-)

**Comment by controversial** [comment]
Great, thanks. I get it now. I've only really used sets for subtraction
before, I had no idea they were so powerful.

**Comment by dgelessus** [comment]
@cclauss To be honest I had no idea which operator (`-` or `|`) takes precedence `;)` It doesn't matter in this case because the first and third set have no items in common. You can subtract set 3 from set 2 and union that with set 1, or you can do the union first and the subtraction second, in the end the result is the same.

@The-Penultimate-Defenestrator If you have any questions, just ask `:)` That PR changed a bunch of things, if there's anything you don't understand or disagree with, please say so. (The messed up diff because of the file moves didn't really help.)

**Comment by controversial** [comment]
Great. Thanks for the clearing-up. I'll take some time to read the code,
and I'll let you know if I have any more questions.

**Pull request merged**